src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,66 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    public static final int MAX_LENGTH = 10;
+    private Object[] keys;
+    private Object[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = new Object[MAX_LENGTH];
+        values = new Object[MAX_LENGTH];
+        size = 0;
+    }
+
+    private void addLength() {
+        Object[] newKeys = new Object[size + MAX_LENGTH];
+        Object[] newValues = new Object[size + MAX_LENGTH];
+
+        for (int i = 0; i < this.size; i++) {
+            newKeys[i] = this.keys[i];
+            newValues[i] = this.values[i];
+        }
+        this.keys = newKeys;
+        this.values = newValues;
+    }
+
+    private int getIndex(K key) {
+        int index = -1;
+        int localSize = this.size == 0 ? 1 : this.size;
+        for (int i = 0; i < localSize; i++) {
+            if ((key == this.keys[i]) || key != null && key.equals(this.keys[i])) {
+                index = i;
```

Comment:
```suggestion
                return i;
```
змінна index зайва, тут можна повертати одразу 'i' а в кінці методу -1